// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package testing_extension_namespace;

import "arolla/proto/testing/test.proto";

extend testing_namespace.Root {
  optional int32 extension_x_int32 = 100000;
  repeated int32 repeated_extension_x_int32 = 100001;
  optional testing_namespace.Root root_reference = 100002;
}

message InnerExtension {
  extend testing_namespace.Inner {
    optional InnerExtension inner_ext = 100000;
  }
  optional int32 inner_extension_x_int32 = 1;
  repeated int32 repeated_inner_extension_x_int32 = 2;
  optional testing_namespace.Root root_reference = 3;
}
